dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguagePluginHandler.java

```diff
@@ -77,7 +79,12 @@ public VersionTypeValidation validateWorkflowSet(Set<SourceFile> sourcefiles, St
             if (mainDescriptor.isPresent()) {
```

Comment:
It was like this, but if there is no mainDescriptor, it seems like you could just throw an exception here.

But there may be a case I'm missing where we want to delegate that error to the plugin, so see what Denis says.

Reply:
I think if there is no descriptor entirely to even parse, that's on us and we should deal with it. 
That said, do we normally throw exceptions if a workflow is entirely missing?
I thought we would just ignore that but could totally be wrong

Reply:
It looks like the existence of the main descriptor should have already been checked [here](https://github.com/dockstore/dockstore/blob/0e348498551dc6f99b91925eab0bc4e62aefbd86/dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoInterface.java#L700) and [here](https://github.com/dockstore/dockstore/blob/0e348498551dc6f99b91925eab0bc4e62aefbd86/dockstore-webservice/src/main/java/io/dockstore/webservice/resources/HostedWorkflowResource.java#L251) before calling the language plugin handler but I will add code to handle a missing main descriptor here in any case.